pkg/agent/openflow/pipeline.go

```diff
@@ -2208,53 +2163,59 @@ func (c *client) genPacketInMeter(meterID binding.MeterIDType, rate uint32) bind
+		createOFTable(c.bridge, sessionAffinityTable, binding.LastTableID, binding.TableMissActionNone)
+		createOFTable(c.bridge, serviceLBTable, endpointDNATTable.GetID(), binding.TableMissActionNext)
+		createOFTable(c.bridge, endpointDNATTable, c.egressEntryTable, binding.TableMissActionNext)
+		createOFTable(c.bridge, conntrackCommitTable, hairpinSNATTable.GetID(), binding.TableMissActionNext)
+		createOFTable(c.bridge, hairpinSNATTable, L2ForwardingOutTable.GetID(), binding.TableMissActionNext)
 	} else {
-		c.pipeline[spoofGuardTable] = bridge.CreateTable(spoofGuardTable, conntrackTable, binding.TableMissActionDrop)
-		c.pipeline[ipv6Table] = bridge.CreateTable(ipv6Table, conntrackTable, binding.TableMissActionNext)
-		c.pipeline[conntrackStateTable] = bridge.CreateTable(conntrackStateTable, dnatTable, binding.TableMissActionNext)
-		c.pipeline[dnatTable] = bridge.CreateTable(dnatTable, c.egressEntryTable, binding.TableMissActionNext)
-		c.pipeline[conntrackCommitTable] = bridge.CreateTable(conntrackCommitTable, L2ForwardingOutTable, binding.TableMissActionNext)
+		createOFTable(c.bridge, spoofGuardTable, conntrackTable.GetID(), binding.TableMissActionDrop)
+		createOFTable(c.bridge, ipv6Table, conntrackTable.GetID(), binding.TableMissActionNext)
+		createOFTable(c.bridge, conntrackStateTable, dnatTable.GetID(), binding.TableMissActionNext)
+		createOFTable(c.bridge, dnatTable, c.egressEntryTable, binding.TableMissActionNext)
+		createOFTable(c.bridge, conntrackCommitTable, L2ForwardingOutTable.GetID(), binding.TableMissActionNext)
 	}
 	// The default SNAT is implemented with OVS on Windows.
 	if c.enableEgress || runtime.IsWindowsPlatform() {
-		c.pipeline[snatTable] = bridge.CreateTable(snatTable, l2ForwardingCalcTable, binding.TableMissActionNext)
+		createOFTable(c.bridge, snatTable, l2ForwardingCalcTable.GetID(), binding.TableMissActionNext)
 	}
 	if runtime.IsWindowsPlatform() {
-		c.pipeline[uplinkTable] = bridge.CreateTable(uplinkTable, spoofGuardTable, binding.TableMissActionNone)
+		createOFTable(c.bridge, uplinkTable, spoofGuardTable.GetID(), binding.TableMissActionNone)
 	}
 	if c.enableAntreaPolicy {
-		c.pipeline[AntreaPolicyEgressRuleTable] = bridge.CreateTable(AntreaPolicyEgressRuleTable, EgressRuleTable, binding.TableMissActionNext)
-		c.pipeline[AntreaPolicyIngressRuleTable] = bridge.CreateTable(AntreaPolicyIngressRuleTable, IngressRuleTable, binding.TableMissActionNext)
+		createOFTable(c.bridge, AntreaPolicyEgressRuleTable, EgressRuleTable.GetID(), binding.TableMissActionNext)
+		createOFTable(c.bridge, AntreaPolicyIngressRuleTable, IngressRuleTable.GetID(), binding.TableMissActionNext)
 	}
 }
 
+// createOFTable sets the missAction and the next table ID of the given table according to the pipeline. Then it creates the table on the bridge so that the table can leverage the bridge to send OpenFlow messages to OVS. At last, it adds the table into the ofTableCache.
```

Comment:
I feel we can remove "so that the table can leverage the bridge to send OpenFlow messages to OVS". A table is not just for sending OF messages right?

Reply:
updated.